File: ros2_ws/src/my_cpp_pkg/src/add_two_ints_server.cpp
#include <string_view>

#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"

using SrvMsg = example_interfaces::srv::AddTwoInts;
using request = example_interfaces::srv::AddTwoInts_Request::SharedPtr;
using response = example_interfaces::srv::AddTwoInts_Response::SharedPtr;

using std::placeholders::_1;
using std::placeholders::_2;

namespace {
std::string_view sv_name{"add_two_ints_server"};
}  // namespace

class AddTwoIntsNode : public rclcpp::Node {
 public:
  AddTwoIntsNode() : Node(sv_name.data()) {
    server_ = this->create_service<SrvMsg>(
        "add_two_ints",
        std::bind(&AddTwoIntsNode::callbackAddTwoInts, this, _1, _2));
    RCLCPP_INFO(this->get_logger(), "%s has started...", sv_name.data());
  }

 private:
  rclcpp::Service<SrvMsg>::SharedPtr server_;
  void callbackAddTwoInts(const request req, const response res) {
    res->sum = req->a + req->b;
    RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", req->a, req->b,
                res->sum);
  }
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<AddTwoIntsNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/add_two_ints_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

node_name = 'add_two_ints_server'

class AddTwoIntsServerNode(Node):
    def __init__(self):
        super().__init__(f'{node_name}')
        self.get_logger().info(f'{node_name} has started...')
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_add_two_ints)

    def callback_add_two_ints(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info(f'{request.a} + {request.b} = {response.sum}')
        return response

def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsServerNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()